python/openinference-instrumentation/src/openinference/instrumentation/_tracers.py

```diff
@@ -152,30 +152,69 @@ def start_span(
         *,
         openinference_span_kind: Optional["OpenInferenceSpanKind"] = None,
     ) -> OpenInferenceSpan:
+        user_attributes = dict(attributes) if attributes else {}
+        span_kind_attributes = (
+            get_span_kind_attributes(openinference_span_kind)
+            if openinference_span_kind is not None
+            else {}
+        )
+        context_attributes = dict(get_attributes_from_context())
+
         otel_span: Span
         if get_value(_SUPPRESS_INSTRUMENTATION_KEY):
             otel_span = INVALID_SPAN
         else:
+            # Apply masking to attributes before passing to sampler to ensure
+            # samplers don't see sensitive data that should be masked
+            combined_attributes = self._get_masked_attributes_for_sampling(
+                {
+                    **user_attributes,
+                    **span_kind_attributes,
+                    **context_attributes,
```

Comment:
context attributes probably should go first otherwise users won't be able to override it
```suggestion
                    **context_attributes,
                    **user_attributes,
                    **span_kind_attributes,
```